baseline-error-prone/src/test/java/com/palantir/baseline/errorprone/ResourceIdentifierGetEqualsUsageTest.java

```diff
@@ -0,0 +1,193 @@
+                        "import com.palantir.ri.ResourceIdentifier;",
+                        "public class Test {",
+                        "  boolean f(ResourceIdentifier rid) {",
+                        "    return rid.getLocator().equals(\"test\");",
+                        "  }",
+                        "}")
+                .addOutputLines(
+                        "Test.java",
+                        "import com.palantir.ri.ResourceIdentifier;",
+                        "public class Test {",
+                        "  boolean f(ResourceIdentifier rid) {",
+                        "    return rid.hasLocator(\"test\");",
+                        "  }",
+                        "}")
+                .doTest();
+    }
+
+    @Test
+    void testHasService() {
+        fix().addInputLines(
+                        "Test.java",
+                        "import com.palantir.ri.ResourceIdentifier;",
+                        "public class Test {",
+                        "  boolean f(ResourceIdentifier rid) {",
+                        "    return rid.hasService(\"test\");",
+                        "  }",
+                        "}")
+                .expectUnchanged()
+                .doTest();
+
+        fix().addInputLines(
+                        "Test.java",
+                        "import com.palantir.ri.ResourceIdentifier;",
+                        "public class Test {",
+                        "  boolean f(ResourceIdentifier rid) {",
+                        "    return rid.getService().equals(\"test\");",
+                        "  }",
+                        "}")
+                .addOutputLines(
+                        "Test.java",
+                        "import com.palantir.ri.ResourceIdentifier;",
+                        "public class Test {",
+                        "  boolean f(ResourceIdentifier rid) {",
+                        "    return rid.hasService(\"test\");",
+                        "  }",
+                        "}")
+                .doTest();
+        fix().addInputLines(
+                        "Test.java",
+                        "import com.palantir.ri.ResourceIdentifier;",
+                        "public class Test {",
+                        "  private static final String SERVICE = \"FOO\";",
+                        "  boolean f(ResourceIdentifier rid) {",
+                        "    return rid.getService().equals(SERVICE);",
+                        "  }",
+                        "}")
+                .addOutputLines(
+                        "Test.java",
+                        "import com.palantir.ri.ResourceIdentifier;",
+                        "public class Test {",
+                        "  private static final String SERVICE = \"FOO\";",
+                        "  boolean f(ResourceIdentifier rid) {",
+                        "    return rid.hasService(SERVICE);",
+                        "  }",
+                        "}")
+                .doTest();
```

Comment:
nit: If this case is important enough to test for `getService`, then we should test it for every method.